src/Behat/Behat/Transformation/Transformer/FromStringObjectTransformer.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Transformation\Transformer;
+
+use Behat\Behat\Definition\Call\DefinitionCall;
+use Behat\Behat\Transformation\Exception\FactoryMethodNotFound;
+use Closure;
+use ReflectionClass;
+use ReflectionParameter;
+
+/**
+ * Transforms typehinted parameters using their `fromString` factory method (if exists).
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+final class FromStringObjectTransformer implements ArgumentTransformer
+{
+    const FACTORY_METHOD = 'fromString';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsDefinitionAndArgument(DefinitionCall $definitionCall, $argumentIndex, $argumentValue)
+    {
+        return !is_object($argumentValue) && null !== $this->getParameterClassByIndex($definitionCall, $argumentIndex);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function transformArgument(DefinitionCall $definitionCall, $argumentIndex, $argumentValue)
+    {
+        $class = $this->getParameterClassByIndex($definitionCall, $argumentIndex);
+
+        if (!$class->hasMethod(self::FACTORY_METHOD)) {
```

Comment:
Probably also need to check the method is compatible i.e. Unary, can take a string as it's arguments (compatible typehints)